packages/synapse-react-client/src/utils/jsonschema/getRequiredAttributes.test.ts

```diff
@@ -0,0 +1,102 @@
+import { describe, it, expect } from 'vitest'
+import getRequiredAttributes from './getRequiredAttributes'
+
+describe('getRequiredAttributes', () => {
+  it('returns empty array for undefined/null/non-object', () => {
+    expect(getRequiredAttributes(undefined)).toEqual([])
+    expect(getRequiredAttributes(null)).toEqual([])
+    // @ts-expect-error intentional wrong type
+    expect(getRequiredAttributes(5)).toEqual([])
+  })
+
+  it('returns required fields from simple schema', () => {
+    const schema = {
+      type: 'object',
+      required: ['a', 'b'],
+      properties: {
+        a: { type: 'string' },
+        b: { type: 'number' },
+        c: { type: 'boolean' },
+      },
+    }
+    expect(getRequiredAttributes(schema).sort()).toEqual(['a', 'b'])
+  })
+
+  it('deduplicates required fields', () => {
+    const schema = {
+      type: 'object',
+      required: ['a', 'b', 'a'],
+    }
+    expect(getRequiredAttributes(schema).sort()).toEqual(['a', 'b'])
+  })
+
+  it('handles allOf union of required', () => {
+    const schema = {
+      allOf: [
+        { required: ['a', 'b'] },
+        { required: ['b', 'c'] },
+        { properties: { d: { type: 'string' } } },
+      ],
+    }
+    expect(getRequiredAttributes(schema).sort()).toEqual(['a', 'b', 'c'])
+  })
+
+  it('handles anyOf union of required', () => {
+    const schema = {
+      anyOf: [{ required: ['x'] }, { required: ['y', 'z'] }, { required: [] }],
+    }
+    expect(getRequiredAttributes(schema).sort()).toEqual(['x', 'y', 'z'])
+  })
```

Comment:
Not a valid test, with this schema, there is technically no requirement.
```suggestion
  it('handles anyOf union of required', () => {
    const schema = {
      anyOf: [{ required: ['x'] }, { required: ['y', 'z'] }, { required: [] }],
    }
    expect(getRequiredAttributes(schema).sort()).toEqual([])
  })
```